storage.go

```diff
@@ -41,15 +40,15 @@ func (s *Storage) createDir(ctx context.Context, path string, opt pairStorageCre
 
 	_, err = s.client.NewDirectoryURL(path).GetProperties(ctx)
 	if err == nil {
-		// The directory exist, we should delete it.
-		// ref: https://github.com/beyondstorage/specs/blob/master/rfcs/134-write-behavior-consistency.md
-		_, err = s.client.NewDirectoryURL(path).Delete(ctx)
-		if err != nil {
-			return nil, err
-		}
+		// The directory exist, we should reset the metadata.
+		_, err = s.client.NewDirectoryURL(path).SetMetadata(ctx, nil)
```

Comment:
Sorry for my inaccurate expression. We don't need to call `SetMetadata` here. We can use the metadata returned by `GetProperties` like `lastModified` to assign the value for `o.lastModified`. Like what we did in https://github.com/beyondstorage/go-service-kodo/blob/master/storage.go#L51-L60